services/github/project/api/routes.py

```diff
@@ -200,7 +210,40 @@ def get_items():
                     item['reviewer_points'] = 0
                 item['num_reactions'] = r.get('reactions').get('totalCount') or 0
                 item['title'] = r.get('title')
-                items.append(item)
+                # check if date is between start and end date
+                if start_creation_date != None and end_creation_date != None:
+                    try:
+                        item_date = datetime.datetime.strptime(item['createdAt'].split('T')[0], "%Y-%m-%d")
+                        start_date = datetime.datetime.strptime(start_creation_date, "%Y-%m-%d")
+                        end_date = datetime.datetime.strptime(end_creation_date, "%Y-%m-%d")
+
+                        if check_between_dates(start_date, end_date, item_date):
+                            items.append(item)
+                    except ValueError:
+                        response_object = {
+                            'status': 'fail',
+                            'message': "date format error, format should be YYYY-MM-DD"
+                        }
+                        db.session.rollback()
+                        return jsonify(response_object), 400
+                elif start_updated_date != None and end_updated_date != None:
+                    try:
+                        item_date = datetime.datetime.strptime(item['updatedAt'].split('T')[0], "%Y-%m-%d")
```

Comment:
Prefer not to do string manipulation here. It just causes problems later when dealing with varying timezones. Better to convert the full string into a full `datetime` object.